src/Tester.ts

```diff
@@ -10,75 +10,369 @@ After the test a test report can be generated and analyzed to get more meaning o
+                    )
+                    container.passed = true
+                    container.subscriptionReport.passed = true
+                    container.subscriptionReport.result = new Result(
+                        100,
+                        "Timeout when subscribing: Due to the design of node-wot this can mean either the subscription was unsuccessful or it was successful but no data was received."
+                    )
+                    break
+                case SubscriptionStatus.Successful:
+                    self.log("* Successfully subscribed to " + interactionSpecifier + ".")
+                    container.subscriptionReport.passed = true
+                    container.subscriptionReport.result = new Result(200)
+                    await Promise.race([Utils.sleepInMs(eventConfig.MsListen), earlyListenTimeout])
+                    break
+            }
+            return
+        }
+
+        /**
+         * Records received data and checks if it is valid.
+         * @param receivedData The received data package.
+         */
+        async function handleReceivedData(receivedData: any): Promise<void> {
+            let receivedTimeStamp = new Date()
+            ++indexOfEventData
+            // Stop recording if maximum number of recorded data packages is reached.
+            if (eventConfig.MaxAmountRecvData != null) {
+                if (indexOfEventData >= eventConfig.MaxAmountRecvData) {
+                    // Stop Listening for Data.
+                    earlyListenTimeout.resolve()
+                    return
+                }
+            }
+            // Stop handling if sent TD does not have "data".
+            if (testMode == Utils.InteractionType.Event && !interaction.hasOwnProperty("data")) {
+                return
+            }
+            self.log("* " + receivedDataMsg + "[index: " + indexOfEventData + "]: " + JSON.stringify(receivedData, null, " "))
+            // Checking if data package has JSON format.
+            try {
+                let temp: JSON = receivedData
+            } catch (jsonError) {
+                self.log("* " + receivedDataMsg + "[index: " + indexOfEventData + "] is not in JSON format")
+                container.passed = false
+                container.eventDataReport.passed = false
+                let result = new Result(15, "* Received data [index: " + indexOfEventData + "] is not in JSON format: " + jsonError)
+                container.eventDataReport.received.push(new EventData(receivedTimeStamp, receivedData, result))
+                return
+            }
+            // Checking if data package validates against its schema.
+            var validationError: Array<any>
+            if (testMode == Utils.InteractionType.Event)
+                validationError = Utils.validateResponse(interactionName, receivedData, self.testConfig.SchemaLocation, Utils.SchemaType.EventData)
```

Comment:
Only realized here. If there are two interactions with the same name, this stuff fails. This is generally not the case so it an assumption can be made

Reply:
I do not think so. The interaction is identified by a tuple of schemaType and name. Per schema type only one name is possible. This is a side effect of my changes to extractSchema(), which were done to be able to be able to support interactions with more than one schema (event -> TD example 35) I did not really think about it but if a property could also have a schema for subscription and cancellation I would suggest adding a propertySubscription and propertyCancellation schemaType in the future. I noticed your described behavior and changed it for getInteractionByName() which no uses a tuple of interactionType and name.

Reply:
I see, that sounds good. For properties, there is no plan to add such information